pkg/resourceprovider/provider.go

```diff
@@ -398,15 +413,41 @@ func (p *resourceProvider) queryBoth(now pmodel.Time, resource schema.GroupResou
 	wg.Wait()
 
 	if cpuErr != nil {
-		return nsQueryResults{
-			namespace: namespace,
-			err:       fmt.Errorf("unable to fetch node CPU metrics: %v", cpuErr),
+		if promErr, ok := cpuErr.(*client.Error); ok {
```

Comment:
ditto: **Actually, I think we should make promClient.Query always return a *client.Error instead**